tests/unit/single_table/test_base.py

```diff
@@ -561,3 +563,990 @@ def test_update_transformers(self):
+            float_rtol=0.01,
+            progress_bar=None,
+            output_file_path=None,
+        )
+
+        # Assert
+        pd.testing.assert_frame_equal(result, sampled_data)
+        rows, conditions, trans_cond, float_rtol, sampled = instance._sample_rows.call_args[0]
+        assert instance._sample_rows.call_count == 2
+
+    def test__sample_batch_storing_output_file(self, tmpdir):
+        """Test that an output file is properly stored while sampling.
+
+        Test that an output file is properly stored as well as that the progress bar is updated.
+        """
+        # Setup
+        sampled_data = pd.DataFrame({
+            'name': ['John', 'Doe', 'John Doe', 'John Doe John'],
+            'salary': [80., 60., 100., 300.]
+        })
+        instance = Mock()
+        instance._sample_rows.side_effect = [
+            (sampled_data, 4),
+            (sampled_data, 5),
+            (sampled_data, 10),
+        ]
+        mock_progress_bar = Mock()
+
+        path = tmpdir / 'file.csv'
+        # Create an empty csv file
+        open(path, mode='a').close()
+
+        # Run
+        result = BaseSynthesizer._sample_batch(
+            instance,
+            batch_size=10,
+            max_tries=100,
+            conditions=None,
+            transformed_conditions=None,
+            float_rtol=0.01,
+            progress_bar=mock_progress_bar,
+            output_file_path=path,
+        )
+
+        # Assert
+        pd.testing.assert_frame_equal(result, sampled_data)
+        assert instance._sample_rows.call_count == 3
+        expected_stored_data = pd.DataFrame({
+            'name': [
+                'John',
+                'Doe',
+                'John Doe',
+                'John Doe John',
+                'John Doe John',
+                'John',
+                'Doe',
+                'John Doe',
+                'John Doe John',
+            ],
+            'salary': [
+                80.,
+                60.,
+                100.,
+                300.,
+                300.,
+                80.,
+                60.,
+                100.,
+                300.,
+            ]
+        })
+        data = pd.read_csv(path)
+        pd.testing.assert_frame_equal(expected_stored_data, data)
+        mock_progress_bar.update.call_count == 3
+
+    def test__make_condition_dfs(self):
+        """Test that the condition dfs are being created as expected."""
+        # Setup
+        condition_a = Condition({'name': 'John Doe'})
+        condition_b = Condition({'salary': 80.})
+
+        # Run
+        result = BaseSynthesizer._make_condition_dfs([condition_a, condition_b])
+
+        # Assert
+        expected_result = [
+            pd.DataFrame({'name': ['John Doe']}),
+            pd.DataFrame({'salary': [80.]}),
+        ]
+
+        for res, exp in zip(result, expected_result):
+            pd.testing.assert_frame_equal(res, exp)
+
+    def test__sample_in_batches(self):
```

Comment:
Need another test where result is empty (to get 100% coverage).

Reply:
👍🏻  done